packs/st2ask/actions/workflows/any_response.yaml

```diff
@@ -0,0 +1,96 @@
+            action: core.local
+            input:
+                cmd: echo "Beginning approval example workflow"
+            on-complete:
+                - basic_approval
+
+
+        # We hit our first approval task. The first time this action runs, the execution context will be empty, so
+        # naturally, this data will fail to validate, so the action will return an `pending` status, and this workflow
+        # will be paused. It will remain in this state until someone passes additional data into the executions API.
+        # When this happens, the action will re-run, and re-attempt to validate the data in the context. If this data
+        # validates, the action will return a success status and will continue with the workflow.
+        #
+        # If it does not validate, it will return to `pending` status.
+        #
+        # Note that the optional "schema" parameter is supplied down in the "ask_for_2fa_token" task, but is omitted here.
+        # By default, `st2.ask` will use the default JSON schema (with a bit of built-in logic) that validates a
+        # simple approve/reject response stored in the execution context.
+        # When this happens, the `st2.ask` will have a bit of conditional logic to detect the approve/reject status
+        # and return a success or failed state.
+        #
+        # So, if someone approves this, such as via an `st2 approve` command, it will continue to task2
+        basic_approval:
+            action: st2.ask
+            input:
+                tag: managers
+                # Note that the "users" and "roles" parameters are omitted here, so anyone can respond to this task
+            on-success:
+                - task2
+
+        task2:
+            action: core.local
+            input:
+                cmd: echo "Got to task1"    
+            on-complete:
+                - ask_for_2fa_token
+
+        # Now, we arrive at our second approval example. We know that one of the upcoming tasks "send_request_with_2fa"
+        # requires a password AND a second factor for authentication. We can provide the password here (or in st2kv of course)
+        # but for the second factor, we need to ask someone to intervene.
+        #
+        # In this instance of `st2.ask`, the same initial treatment applies. The execution context is empty,
+        # so naturally, it will fail to validate, and the action will return `pending` status.
+        #
+        # Note that we're providing our own JSON schema to the `schema` parameter here. `st2.ask` will
+        # use this to validate data stored in the execution context instead of the default schema that
+        # validates a simple approve/reject response from the user.
+        # When we do this, the action doesn't have a way of knowing what a "rejected" approval means,
+        # so this will block in `pending` status until someone sends data into the context that will validate
+        # the schema.
+        ask_for_2fa_token:
```

Comment:
I had this in there to illustrate the difference between a user-provided schema and the default approve/reject schema, but based on other comments it seems we're moving away from that - so I'll remove this